flexmeasures/data/models/planning/storage.py

```diff
@@ -645,19 +681,30 @@ def compute(self, skip_validation: bool = False) -> SchedulerOutputType:
 
         # Obtain the storage schedule from all device schedules within the EMS
         storage_schedule = ems_schedule[0]
+        soc_schedule = integrate_time_series(storage_schedule, soc_at_start)
```

Comment:
This is still missing a few parameters. Specifically, the exact conversion efficiencies and storage efficiency that was also passed to the `device_scheduler` (as part of the the `device_constraints` I think).  Without these, the computed SoC schedule will come out differently.

Alternatively, the correct SoC schedule is also already contained in the `scheduler_results` or the model (`_` in line 671) or both, but might still be scaled (as a function of the event resolution) and translated (as a function of `soc_at_start`).